src/components/PeerGroupsUpdate.tsx

```diff
@@ -0,0 +1,193 @@
+import React, {useEffect, useRef, useState} from 'react';
+import {useDispatch, useSelector} from "react-redux";
+import {RootState} from "typesafe-actions";
+import { actions as peerActions } from '../store/peer';
+import {
+    Col,
+    Row,
+    Typography,
+    Space,
+    Button, Drawer, Form, Select, Tag, Divider
+} from "antd";
+import type { CustomTagProps } from 'rc-select/lib/BaseSelect'
+import {useAuth0} from "@auth0/auth0-react";
+import {PeerGroupsToSave} from "../store/peer/types";
+import {Group, GroupPeer} from "../store/group/types";
+
+const { Paragraph } = Typography;
+const { Option } = Select;
+
+const PeerGroupsUpdate = () => {
+    const { getAccessTokenSilently } = useAuth0()
+    const dispatch = useDispatch()
+    const groups =  useSelector((state: RootState) => state.group.data)
+    const peer =   useSelector((state: RootState) => state.peer.peer)
+    const updateGroupsVisible = useSelector((state: RootState) => state.peer.updateGroupsVisible)
+    const savedGroups = useSelector((state: RootState) => state.peer.savedGroups)
+
+    const [tagGroups, setTagGroups] = useState([] as string[])
+    const [selectedTagGroups, setSelectedTagGroups] = useState([] as string[])
+    const [peerGroups, setPeerGroups] = useState([] as GroupPeer[])
+    const [peerGroupsToSave, setPeerGroupsToSave] = useState({
+        ID: '',
+        groupsNoId: [],
+        groupsToSave: [],
+        groupsToRemove: [],
+        groupsToAdd: []
+    } as PeerGroupsToSave)
+
+    const [form] = Form.useForm()
+
+    useEffect(() => {
+        if (!peer) return
+        const gs = peer?.Groups?.map(g => ({ID: g?.ID || '', Name: g.Name} as GroupPeer)) as GroupPeer[]
+        const gs_name = gs?.map(g => g.Name) as string[]
+        setPeerGroups(gs)
+        setSelectedTagGroups(gs_name)
+        form.setFieldsValue({
+            groups: gs_name
+        })
+    }, [peer])
+
+    useEffect(() => {
+        setTagGroups(groups?.map(g => g.Name) || [])
+    }, [groups])
+
+    useEffect(() => {
+        const groupsToRemove = peerGroups.filter(pg => !selectedTagGroups.includes(pg.Name)).map(g => g.ID)
+        const groupsToAdd = (groups as Group[]).filter(g => selectedTagGroups.includes(g.Name) && !groupsToRemove.includes(g.ID || '') && !peerGroups.find(pg => pg.ID === g.ID)).map(g => g.ID) as string[]
+        const groupsNoId = selectedTagGroups.filter(stg => !groups.find(g => g.Name === stg))
+        setPeerGroupsToSave({
+            ...peerGroupsToSave,
+            ID: peer?.ID || '',
+            groupsToRemove,
+            groupsToAdd,
+            groupsNoId
+        })
+    }, [selectedTagGroups])
```

Comment:
Codacy has a fix for the issue: [Replace `····},·[selectedTagGroups])` with `··},·[selectedTagGroups]);`](https://app.codacy.com/gh/netbirdio/dashboard/pullRequest?prid=9560373)
```suggestion
  }, [selectedTagGroups]);
```